Feat: List 구현

zockq1 opened the pull request:
1. list-item 디렉토리에 테마, 배너, 카드, 뱃지 각각 구현 했습니다.
2. post 디렉토리에 게시글 목록 구현했습니다.
3. badge 디렉토리에 사용중 | 미사용 상태 뱃지 구현했습니다

zockq1 committed:
Feat: 사용중 | 미사용 상태 뱃지 구현

zockq1 committed:
feat: 게시글 목록 구현

zockq1 committed:
Feat: 배너, 뱃지, 테마, 카드 구현

zockq1 committed:
Fix: margin 태그 위치 수정

zockq1 committed:
Chore: 스토리북 폴더 구조 변경

zockq1 committed:
Feat: 유저 의견 구현

zockq1 committed:
Chore: 스토리북 버튼 폴더 구조 추가

heegu0311 commented on src/share/ui/badge/StatusBadge.tsx:
```diff
@@ -0,0 +1,34 @@
+import { cn } from '@/utils/cn';
+
+export const StatusBadgeVariants = cva('', {
+  variants: {
```
text-center로 하고
부모 컴포넌트의 너비에 따라 full 과 같은 boolean 속성을 추가하는게 어떨까요?
아래 부분처럼 너비가 조금씩 달라지는 애들도 있네요!

<img width="292" alt="image" src="https://github.com/zockq1/piggymy-frontend-admin/assets/13115713/682b1db6-a3ff-45a8-a106-e55e836f8789">

zockq1 replied:
text-center 적용 하고,HTMLAttributes<HTMLDivElement> 을 extents 해서
사용할 때 <StatusBadge className="w-16"> 처럼 tailwind를 이용해 더 자유롭게 조절 가능하도록 바꿔봤습니다

heegu0311 commented on src/share/ui/post/PostList.tsx:
```diff
@@ -0,0 +1,39 @@
+          >
+            <Link
+              href={route}
+              className="inline-block truncate text-[15px] font-bold"
```
inline-block을 명시하신 이유가 따로 있을까요? 제거 해보니 동일하게 동작하는 것 같아서요 ㅎ

heegu0311 commented on src/share/ui/post/PostList.tsx:
```diff
@@ -0,0 +1,39 @@
+            >
+              {title}
+            </Link>
+            <span className="inline-block divide-x text-end text-[9px]">
```
28번 줄로 바꾸면 좀더 디자인과 비슷해집니다!

```tsx
<span className="h-2 divide-x divide-[#d9d9d9] text-end text-[9px]">
```

heegu0311 commented on src/share/ui/list-item/Add.tsx:
```diff
@@ -0,0 +1,53 @@
+      href={route}
+      className={cn(
+        BadgeVariants({ type, status: isSelected ? 'selected' : 'unselected' }),
+        'm-4 flex items-center justify-center',
```
`m-4`이 필요한지 한번더 검토부탁드립니다~
저는 빼도 되지 않을까 싶어서요 ㅎ

heegu0311 commented on src/share/ui/list-item/Badge.tsx:
```diff
@@ -0,0 +1,58 @@
+interface BadgeProps {
+  title: string;
+  image: string | StaticImageData;
+  registrationDate: string;
```
날짜 관련해서는 moment.js를 도입해보는 것을 제안드립니다!
string으로 처리하던 것들이 어느 순간 번거롭더라구요.

heegu0311 commented on src/share/ui/list-item/Badge.tsx:
```diff
@@ -0,0 +1,58 @@
+  route,
+}: BadgeProps) {
+  return (
+    <div className="mx-4 flex h-[199px] w-[170px] flex-col justify-between">
```
`mx-4`이 필요한지 한번더 검토부탁드립니다~
저는 빼도 되지 않을까 싶어서요 ㅎ

slide 같은 형태로 구현하거나, flex로 처리하게되면 gap 같은걸로 조정하면 될듯합니다
<img width="392" alt="image" src="https://github.com/zockq1/piggymy-frontend-admin/assets/13115713/08f99dae-91af-4ba9-b7b3-f228b515606a">

heegu0311 reviewed (approved):
@zockq1 코드가 다들 비슷한 패턴으로 작성되어서 적응되니 잘 읽히는 것 같습니다!

날짜 관련 텍스트에 moment.js 사용을 제안드리는 것 이외 마이너한것들 몇개 적었습니다.
고생하셨습니다 :)

모두 당장은 재민님 선택 사항이라 판단해서 Approve 해두겠습니다.

zockq1 committed:
Fix: 피드백 적용

1. 전부 use client로 변경
2. 마진 제거
3. PostList 디자인 변경
4. 상태 뱃지 클래스명으로 디자인 면경 가능하도록 수정

zockq1 committed:
Refactor: string -> dayjs로 변경

zockq1 committed:
Merge branch 'develop' into feature/list
